app/controllers/CheckYourAnswersController.scala

```diff
@@ -35,10 +38,37 @@ class CheckYourAnswersController @Inject()(
   def onPageLoad(): Action[AnyContent] = cc.authAndGetData() {
     implicit request =>
 
+      val thisPage = CheckYourAnswersPage
+      val waypoints = EmptyWaypoints
+
+      val vatRegistrationDetailsList = SummaryListViewModel(
+        rows = Seq(
+          VatRegistrationDetailsSummary.rowBusinessName(request.userAnswers),
+          VatRegistrationDetailsSummary.rowIndividualName(request.userAnswers),
+          VatRegistrationDetailsSummary.rowPartOfVatUkGroup(request.userAnswers),
+          VatRegistrationDetailsSummary.rowUkVatRegistrationDate(request.userAnswers),
+          VatRegistrationDetailsSummary.rowBusinessAddress(request.userAnswers)
+        ).flatten
+      )
+
+      val hasTradingNameSummaryRow = HasTradingNameSummary.row(request.userAnswers, waypoints, thisPage)
```

Comment:
Naming wise starting option type names with maybe makes things a bit easier to comprehend in PR's and IDE. For futures eventual prefix, either aOrB. If you use intellij's rename option, introduce variable it defaults to these naming concepts.

Monads  have some mental overhead so it helps with understanding the result type  of the map operations and whether things can be flattened etc. 

As the name hasTradingNameSummaryRow is now freed up it can be used clearly in the map operation.

```
maybeHasTradingNameSummaryRow.map { hasTradingNameSummaryRow =>

}
```

Sounds picky, just makes reading easier. Intellij's naming suggestions and greedy autocompletion (completes on any combination of typed characters in sequence) negates most of the typing overhead.